OpenRA.Mods.Common/Traits/Buildings/FreeActor.cs

```diff
@@ -28,25 +28,49 @@ public class FreeActorInfo : ITraitInfo
 		[Desc("Which direction the unit should face.")]
 		public readonly int Facing = 0;
 
-		public virtual object Create(ActorInitializer init) { return new FreeActor(init, this); }
+		[Desc("Whether another actor should spawn upon re-enabling the trait.")]
+		public readonly bool AllowRespawn = false;
+
+		public override object Create(ActorInitializer init) { return new FreeActor(init, this); }
 	}
 
-	public class FreeActor
+	public class FreeActor : ConditionalTrait<FreeActorInfo>
 	{
-		public FreeActor(ActorInitializer init, FreeActorInfo info)
+		FreeActorInfo info;
```

Comment:
The info is exposed via the `ConditionalTrait` superclass (as `Info`).  This and the assignment in the constructor can be removed.